### PR TITLE
Merge latest Develop into Main

### PR DESCRIPTION
**Pull request by grantjllnl**
- Many changes to downloading and post-processing as file/directory structure stabilize
- Introduce DuckDB as the primary mechanism for pulling data from parquet into pandas/notebooks

### CODE REVIEW

File: rawtorolling.py [hunk 1]
```diff
@@ -34,7 +35,7 @@ def main():
     try:
         logging.basicConfig(level=args.log_level,format='%(asctime)s %(message)s', datefmt='%m/%d/%Y %I:%M:%S %p')
     except ValueError:
-        logging.error('Invalid log level: {}'.format(args.log_level))
+        logging.error(f'Invalid log level: {args.log_level}')
```
Comment (lindsey-w) [inline rawtorolling.py:38]: (just out of curiosity) - this doesn't give you a warning? For some reason when I use f-strings in logging messages it always tells me to pass in the args.

File: notebookutil.py [hunk 2]
```diff
@@ -4,34 +4,45 @@
...
 import os
-
 import altair as alt
 import pandas as pd
 from dotenv import load_dotenv
 from ipyfilechooser import FileChooser
-from IPython.display import display
+from IPython import get_ipython
 
+# Import packages used in notebooks
```
Comment (lindsey-w) [inline notebookutil.py:13]: we have to import these here and within the notebook context?

**Review by lindsey-w — APPROVED** [review]
🥳

File: rawtorolling.py (hunk 1 above)
Comment (grantjllnl) [reply rawtorolling.py:38]: Seems ok:

`(wintap-pyutil) [fusioncuisine@spk16 Wintap-PyUtil]$ python rawtorolling.py -d /data/fusioncuisine/wintapv6/llnl_herd/ -s 20240624 -e 20240701 -l BAD
07/10/2023 10:28:10 AM Invalid log level: BAD
`

File: notebookutil.py (hunk 2 above)
Comment (grantjllnl) [reply notebookutil.py:13]: No, as of now, using `%run` in a notebook will import those into the notebooks globals. However, in Dan's refactoring for packaging, he'd prefer to just explicitly have users do the imports. We'll discuss that more as packaging evolves.

**Pull request merged**